rest_framework_simplejwt/authentication.py

```diff
@@ -131,7 +131,8 @@ def get_user(self, validated_token: Token) -> AuthUser:
         except self.user_model.DoesNotExist:
             raise AuthenticationFailed(_("User not found"), code="user_not_found")
 
-        if not user.is_active:
+        # Ensure authentication rule passes
+        if not api_settings.USER_AUTHENTICATION_RULE(user):
```

Comment:
we want to only validate the user is active, not run the authentication rule, since this function is just looking to fetch a given user.

We can revisit this if you open a GitHub issue

Reply:
After thinking about it harder i see how it could cause a problem if a user made their own rule, addressed in latest commit.